tests/test_rewrite_rules/test_rewrite_rules.py

```diff
@@ -32,141 +29,134 @@ class TestEntry:
+        TestEntry(linkml + "context.jsonld", "linkml-model/linkml_model/jsonld/context.jsonld"),
+        TestEntry(linkml + "contextn.jsonld", "linkml-model/contextn.jsonld"),
+    ]
+    meta_vocab_entries: List[TestEntry] = [
+        TestEntry(meta.Element, "linkml-model/docs/meta/Element"),
+        TestEntry(meta.slot, "linkml-model/docs/meta/slot"),
+        TestEntry(meta.Element, "linkml-model/linkml_model/model/schema/meta/Element.yaml/Element.yaml", "text/yaml"),
+        TestEntry(meta.Element, "linkml-model/linkml_model/rdf/meta/Element.ttl/Element.ttl", "text/turtle"),
+        TestEntry(meta.Element, "linkml-model/linkml_model/json/meta/Element.json/Element.json", "application/json"),
+        TestEntry(meta.Element, "linkml-model/linkml_model/shex/meta/Element.shex/Element.shex", "text/shex"),
+        TestEntry(meta.Element, "linkml-model/meta/Element", "text/foo"),
+    ]
 
-        def ev_al(entry: TestEntry) -> bool:
-            if not entry.accept_header:
-                return test_it(entry, default_header)
-            else:
-                r1 = test_it(entry, entry.accept_header)
-                return test_it(entry, entry.accept_header + "," + default_header) and r1
+    return {
+        "meta_entries": meta_entries,
+        "vocab_entries": vocab_entries,
+        "meta_model_entries": meta_model_entries,
+        "meta_vocab_entries": meta_vocab_entries,
+    }
 
-        self.assertTrue(all([ev_al(entry) for entry in entries]))
 
-    def test_type_meta(self):
-        self.rule_test(self.tests.meta_entries)
+@pytest.fixture
+def fail_on_error():
+    return os.environ.get("FAIL_ON_ERROR", "False") == "True"
 
-    def test_type_entry(self):
-        self.rule_test(self.tests.vocab_entries)
 
-    def test_meta_meta(self):
-        self.rule_test(self.tests.meta_model_entries)
+@pytest.fixture(scope="class")
+def results() -> Set[Tuple[str, str, str]]:
+    return set()  # from, to, format
 
-    def test_meta_entry(self):
-        self.rule_test(self.tests.meta_vocab_entries)
 
+def record_results(results, from_url: str, accept_header, to_url: str) -> None:
+    results.add((from_url, to_url, accept_header.split(",")[0]))
 
-if __name__ == "__main__":
-    RewriteRuleTestCase.SERVER = os.environ.get("SERVER", RewriteRuleTestCase.SERVER)
-    unittest.main()
+
+fixture_lists = generate_fixture_lists()
+
+
+@pytest.mark.network
+@pytest.mark.skipif(SKIP_REWRITE_RULES, reason=SKIP_REWRITE_RULES_REASON)
+@pytest.mark.parametrize(
+    "entries",
+    [
+        pytest.param(fixture_lists["meta_entries"], id="meta_entries"),
+        pytest.param(fixture_lists["vocab_entries"], id="vocab_entries"),
+        pytest.param(fixture_lists["meta_model_entries"], id="meta_model_entries"),
+        pytest.param(fixture_lists["meta_vocab_entries"], id="meta_vocab_entries"),
+    ],
+)
+def test_rewrite_rules(entries: List[TestEntry], results, fail_on_error) -> None:
+    def test_it(e: TestEntry, accept_header: str) -> bool:
```

Comment:
why the closure here? also unsure why we are collections results with `record_results` - that's what pytest does for us if we just fail on the cases that fail :)

Reply:
These functions were also nested in the original code. I limited the scope of my changes not knowing if it was an intended stylistic convention and to keep the VSCode diff clean. I did consider removing the record_results and still can, but wanted to check w/ team first.

Reply:
i should have waited to leave comments until i could do a close read, sorry. will do that shortly. agreed on the scope of the PR being a pytest transformation and understood leaving things in place bc unsure of why they are that way - i do the same in this package frequently.

Reply:
@sneakers-the-rat Much appreciated details on the related tests in linkml-runtime! I also sent the PR while rushing out the door and should have made it a draft per the process in contributing.md and double checked the print->logging requirement.

Will catch up on this today, but current thoughts are I'll fix the prints, maybe remove record_results and then copy your notes for the remaining work to a new issue if you don't beat me to it.